zerver/views/users.py

```diff
@@ -425,6 +426,8 @@ def create_user_backend(request: HttpRequest, user_profile: UserProfile,
                           {'email': email})
     except DisposableEmailError:
         return json_error(_("Disposable email addresses are not allowed in this organization"))
+    except EmailAliasNotAllowedForRealm:
+        return json_error(_("Email aliases that make use of + are not allowed."))
```

Comment:
We need a better error message for this.  Maybe: "Email addresses containing `+` are not allowed in this organization."

@rishig FYI

Reply:
lgtm.
From a code reading perspective, it's a bit confusing that the error is EmailAliasNotAllowedForRealm rather than EmailContainsPlusError or something, but didn't spend much time reading the rest of the code.